### PR TITLE
Clarify documentation of (hp::)?DoFHandler::distribute_dofs().

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/dofs/dof_handler.h [hunk 1]
```diff
@@ -370,20 +370,42 @@ class DoFHandler  :  public Subscriptor
...
-   * sequentially enumerate all degrees of freedom. In other words, while the
-   * mesh and the finite element object by themselves simply define a finite
-   * element space $V_h$, the process of distributing degrees of freedom makes
-   * sure that there is a basis for this space and that the shape functions of
-   * this basis are enumerated in an indexable, predictable way.
+   * Go through the triangulation and "distribute" the degrees of
+   * freedoms needed for the given finite element. "Distributing"
+   * degrees of freedom involves allocating memory to store the
+   * indices on all entities on which degrees of freedom can be
+   * located (e.g., vertices, edges, faces, etc) and to then enumerate
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_handler.h]: While you've touched this line, we could take the opportunity to make a pedantic grammatical correction:
etc -> etc.

File: include/deal.II/dofs/dof_handler.h [hunk 2]
```diff
@@ -370,20 +370,42 @@ class DoFHandler  :  public Subscriptor
...
+   * predictable way.
    *
-   * The purpose of this function is first discussed in the introduction to
-   * the step-2 tutorial program.
+   * The exact order in which degrees of freedom on a mesh are
+   * ordered, i.e., the order in which basis functions of the finite
+   * element space are enumerated, is something that deal.II treats as
+   * an implementation detail. By and large, degrees of freedom are
+   * enumerated in the same order in which we traverse cells, but you
+   * should not rely an any specific numbering. In contrast, if you
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_handler.h]: rely **on** any

File: include/deal.II/hp/dof_handler.h [hunk 3]
```diff
@@ -280,25 +280,36 @@ namespace hp
...
      * predictable way.
      *
-     * The purpose of this function is first discussed in the introduction to
-     * the step-2 tutorial program.
+     * The exact order in which degrees of freedom on a mesh are
+     * ordered, i.e., the order in which basis functions of the finite
+     * element space are enumerated, is something that deal.II treats as
+     * an implementation detail. By and large, degrees of freedom are
+     * enumerated in the same order in which we traverse cells, but you
+     * should not rely an any specific numbering. In contrast, if you
```
Comment (jppelteret) [inline include/deal.II/hp/dof_handler.h]: rely **on** any

File: include/deal.II/hp/dof_handler.h [hunk 4]
```diff
@@ -280,25 +280,36 @@ namespace hp
...
-     * sequentially enumerate all degrees of freedom. In other words, while
-     * the mesh and the finite element object by themselves simply define a
-     * finite element space $V_h$, the process of distributing degrees of
-     * freedom makes sure that there is a basis for this space and that the
-     * shape functions of this basis are enumerated in an indexable,
+     * Go through the triangulation and "distribute" the degrees of
+     * freedoms needed for the given finite element. "Distributing"
+     * degrees of freedom involves allocating memory to store the
+     * indices on all entities on which degrees of freedom can be
+     * located (e.g., vertices, edges, faces, etc) and to then enumerate
```
Comment (jppelteret) [inline include/deal.II/hp/dof_handler.h]: etc -> etc.

**Comment by bangerth** [comment]
Thanks -- all fixed.

**Pull request merged**